toxav/toxav.c

```diff
@@ -493,131 +599,37 @@ bool toxav_call_control(ToxAV *av, uint32_t friend_number, Toxav_Call_Control co
 
     switch (control) {
         case TOXAV_CALL_CONTROL_RESUME: {
-            /* Only act if paused and had media transfer active before */
-            if (call->msi_call->self_capabilities == 0 &&
-                    call->previous_self_capabilities) {
-
-                if (msi_change_capabilities(call->msi_call,
-                                            call->previous_self_capabilities) == -1) {
-                    rc = TOXAV_ERR_CALL_CONTROL_SYNC;
-                    goto RETURN;
-                }
-
-                rtp_allow_receiving(call->audio_rtp);
-                rtp_allow_receiving(call->video_rtp);
-            } else {
-                rc = TOXAV_ERR_CALL_CONTROL_INVALID_TRANSITION;
-                goto RETURN;
-            }
-
+            call_control_handle_resume(call, &rc);
```

Comment:
If all these handle functions return their error code, OK on success, we don't need to use out-parameters here. That seems a bit cleaner to me.